### ISSUE TITLE
Late diff decorations in rulers and diff editor

### ISSUE THREAD

**Issue by chrmarti** [Microsoft/vscode#16953]
- VSCode Version: Code - Insiders 1.8.0-insider (25dc9851ef6d41fb4939a37312590cadaee90327, 2016-12-08T21:42:14.353Z)
- OS Version: Darwin x64 15.6.0
- Extensions:

|Extension|Author|Version|
|---|---|---|
|EditorConfig|EditorConfig|0.3.4|
|autofilename|JerryHong|0.1.3|
|vscode-docker|PeterJausovec|0.0.7|
|regex|chrmarti|0.0.7|
|vscode-eslint|dbaeumer|1.1.0|
|ghlink|dt|0.0.4|
|tslint|eg2|0.6.0|
|vscode-scss|mrmlnc|0.6.0|
|PowerShell|ms-vscode|0.7.2|
|debugger-for-chrome|msjsdiag|2.2.2|
|diff|rafaelmaiolla|0.0.1|
|vscode-open-in-github|sysoev|1.2.1|

---

It took several minutes for the annotations to show up. It reproduces on window reload. Note that the 4 lines starting with copyLinesUp have changes that are not decorated properly:
![screen shot 2016-12-08 at 3 55 17 pm](https://cloud.githubusercontent.com/assets/9205389/21032605/5ca14b3e-bd5f-11e6-94e4-c4ade065692c.png)

I also didn't see change annotations in the rulers. Not sure if this might depend on an external system, but since the revision history is local, I would expect it not to.

**Comment by bgashler1** [comment]
I can reproduce.  Does not happen in Stable in the same repo.

![screen shot 2016-12-08 at 5 58 20 pm](https://cloud.githubusercontent.com/assets/11839736/21034875/ff65171e-bd6f-11e6-96c3-5055159f31b5.png)

My devtools console:
```
Failed to load resource: net::ERR_FILE_NOT_FOUND
loader.js:136 Loading "vs/nls" failedt @ loader.js:136
loader.js:137 Detail:  DOMException: Failed to execute 'importScripts' on 'WorkerGlobalScope': The script at 'file:///Applications/Visual%20Studio%20Code%20-%20Insiders.app/Contents/Resources/app/out/vs/base/worker/vs/nls.js' failed to load.(…)t @ loader.js:137
loader.js:139 Error: Failed to execute 'importScripts' on 'WorkerGlobalScope': The script at 'file:///Applications/Visual%20Studio%20Code%20-%20Insiders.app/Contents/Resources/app/out/vs/base/worker/vs/nls.js' failed to load.
    at Error (native)
    at e._load (file:///Applications/Visual%20Studio%20Code%20-%20Insiders.app/Contents/Resources/app/out/vs/base/worker/workerMain.js:4:30284)
    at file:///Applications/Visual%20Studio%20Code%20-%20Insiders.app/Contents/Resources/app/out/vs/base/worker/workerMain.js:4:30114t @ loader.js:139
loader.js:141 Here are the modules that depend on it:t @ loader.js:141
loader.js:142 Array[1]t @ loader.js:142
file:///Applications/Visual%20Studio%20Code%20-%20Insiders.app/Contents/Resources/app/out/vs/editor/common/services/editorSimpleWorker.js Failed to load resource: net::ERR_FILE_NOT_FOUND
simpleWorker.ts:289Worker failed to load vs/editor/common/services/editorSimpleWorker
```

**Comment by alexandrudima** [comment]
The root cause is:
* when bundling the editor worker code, it all gets bundled in a self-contained manner in `vs/base/worker/workerMain`
* I don't want to have nls strings in there (since no UI strings are part of the API regarding what the worker can do), so I don't bundle `vs/nls.js` in that file
* This is a good thing, as it reduces the disk footprint of both the monaco editor and of vscode, especially since we kill/start the web worker often
* What is a bad thing is that there are no build-time checks that no vs/nls dependent code makes its way in there (it needs more work in our build scripts)

TL;DR editor web worker fails loading due to it dragging in vs/nls, and it then falls back to loading the code on the UI thread, computing word suggestions, diff, etc. in the main thread.

**Comment by alexandrudima** [comment]
Created #16967 to track the addition of a build-time check for this.
